Repo: integration tests fail on an empty package directory

### Suggestion

```shell
mkdir packages/gotcha
yarn test-integration
```

```shell
$ yarn test-integration
yarn run v1.22.19
$ yarn jest -c ./tests/integration/jest.config.js
$ /Users/josh/repos/typescript-eslint/node_modules/.bin/jest -c ./tests/integration/jest.config.js
Error: Cannot find module '/Users/josh/repos/typescript-eslint/packages/gotcha/package.json'
Require stack:
- /Users/josh/repos/typescript-eslint/tests/integration/pack-packages.ts
- /Users/josh/repos/typescript-eslint/tests/integration/jest.config.js
- ...
```

This is because `pack-packages.ts` attempts to require the `package.json` in each `packages/*` directory. If the directory is blank, such as from switching between branches that CRUD package names, it throws.

--- a/tests/integration/pack-packages.ts
+++ b/tests/integration/pack-packages.ts
@@ -30,10 +30,12 @@ const tarFolder = tmp.dirSync({
 const tseslintPackages: PackageJSON['devDependencies'] = {};
 for (const pkg of PACKAGES) {
   const packageDir = path.join(PACKAGES_DIR, pkg);
-  const packageJson: PackageJSON = require(path.join(
-    packageDir,
-    'package.json',
-  ));
+  const packagePath = path.join(packageDir, 'package.json');
+  if (!fs.existsSync(packagePath)) {
+    continue;
+  }
+
+  const packageJson = require(packagePath) as PackageJSON;
   if (packageJson.private === true) {
     continue;
   }

